lib/python/picongpu/picmi/lasers/dispersive_pulse_laser.py

```diff
@@ -0,0 +1,158 @@
+        self,
+        waist,
+        wavelength,
+        duration,
+        propagation_direction,
+        polarization_direction,
+        focal_position,
+        centroid_position,
+        a0=None,
+        E0=None,
+        phi0: float = 0.0,
+        picongpu_polarization_type=(PolarizationType.LINEAR),
+        picongpu_spectral_support: float = 6.0,
+        picongpu_sd_si: float = 0.0,
+        picongpu_ad_si: float = 0.0,
+        picongpu_gdd_si: float = 0.0,
+        picongpu_tod_si: float = 0.0,
+        # make sure to always place Huygens-surface inside PML-boundaries,
+        # default is valid for standard PMLs
+        # @todo create check for insufficient dimension
+        # @todo create check in simulation for conflict between PMLs and
+        # Huygens-surfaces
+        picongpu_huygens_surface_positions: typing.List[typing.List[int]] = [
+            [16, -16],
+            [16, -16],
+            [16, -16],
+        ],
+        **kw,
+    ):
+        if waist <= 0:
+            raise ValueError(f"waist must be > 0. You gave {waist=}.")
+        if wavelength <= 0:
+            raise ValueError(f"wavelength must be > 0. You gave {wavelength=}.")
+        if duration <= 0:
+            raise ValueError(f"laser pulse duration must be > 0. You gave {duration=}.")
+
+        self.waist = waist
+        self.wavelength = wavelength
+        self.k0 = 2.0 * math.pi / wavelength
+        self.duration = duration
+        self.focal_position = focal_position
+        self.centroid_position = centroid_position
+        self.propagation_direction = propagation_direction
+        self.polarization_direction = polarization_direction
+        self.a0, self.E0 = self._compute_E0_and_a0(self.k0, E0, a0)
+        self.phi0 = phi0
+
+        self.picongpu_polarization_type = picongpu_polarization_type
+        self.picongpu_spectral_support = picongpu_spectral_support
+        self.picongpu_sd_si = picongpu_sd_si
+        self.picongpu_ad_si = picongpu_ad_si
+        self.picongpu_gdd_si = picongpu_gdd_si
+        self.picongpu_tod_si = picongpu_tod_si
+        self.picongpu_huygens_surface_positions = picongpu_huygens_surface_positions
+
+    def get_as_pypicongpu(self) -> laser.DispersivePulseLaser:
+        self._validate_common_properties()
+        assert (
+            self._propagation_connects_centroid_and_focus()
+        ), "propagation_direction must connect centroid_position and focus_position"
+        pypicongpu_laser = laser.DispersivePulseLaser()
+        pypicongpu_laser.wavelength = self.wavelength
+        pypicongpu_laser.waist = self.waist
+        pypicongpu_laser.duration = self.duration
+        pypicongpu_laser.focus_pos = self.focal_position
+        pypicongpu_laser.phase = self.phi0
+        pypicongpu_laser.E0 = self.E0
+
+        pypicongpu_laser.pulse_init = max(
+            -2 * self.centroid_position[1] / (self.propagation_direction[1] * constants.c) / self.duration,
+            15,
```

Comment:
Please get rid of these ugly hard coded defaults of INIT = 15. 
These have already been removed by a previous PR.

Reply:
I like your `_compute_pulse_init()` solution @chillenzer